service_contracts.rst

```diff
@@ -31,6 +31,109 @@ ServiceRegistry
 
 The ServiceRegistry provides a registry in which services have to register before becoming a full part of the Raiden services system. Services have to deposit RDN tokens in this contract for a successful registration. This avoids attacks using a large number of services and increases the incentive for service provider to not harm the Raiden ecosystem.
 
+Requirements
+------------
+
+- Need to stake for these slots by the Raiden Service Providers (RSPs), but there will be no slashing
+- RSPs must be allowed to withdraw the deposited stake after their registration ends
+- An "auction" mechanism that allows for competition over those slots
+- There must be a way for new RSPs to get a slot
+- We want an easy MVP to get real user feedback after our release
+- Need to provide a way of setting the minimal amount of deposits - the cost of becoming an RSP should be high enough to deter users from double-spending IOUs
+- We want to reduce the amount of work to be done, so try to avoid periodical deployments and / or coordination stakeholders
+- There must be a way for other smart contracts to check if an address is registered as a service provider
+- Avoid new implementations if possible
+- The deployer should be able to choose whether it wants to control the parameters
+
+Design
+------------
+
+In order to avoid periodical deployments, ServiceRegistry does not have a deadline for joining.  ServiceRegistry allows new deposits anytime.
+The required amount of deposit (called the price) changes over time so as to regulate the inflow of the deposits.
+The price movement depends only on the timing of incoming deposits.
+Upon receiving new deposit, the price jumps by some ratio (called the bump ratio).
+Otherwise the price decreases slowly following some exponential decay, dictated by the decay constant.
+
+Use cases
+------------
+
+- Somebody who wants to provide services registers themselves as a RSP.
+- Service smart contracts look up whether an Ethereum address is registered as a RSP.
+- A registered RSP posts a URL.
+- Users of services look up URLs of registered RSPs.
+- Users of services choose one of the registered RSPs randomly.
+
+
+States
+------------
+
+1. Before the deployment transaction, the address of ServiceRegistry has no code, and zero nonce.  The address might already control some ETH or tokens (which nobody can take out).
```

Comment:
Isn't this true for all contracts?

Reply:
I'll take this step away.  This is not important or informative.